src/hotspot/cpu/riscv/riscv.ad

```diff
@@ -5529,17 +5620,39 @@ instruct compareAndExchangeB(iRegINoSp res, indirect mem, iRegI_R12 oldval, iReg
   ins_pipe(pipe_slow);
 %}
 
-instruct compareAndExchangeS(iRegINoSp res, indirect mem, iRegI_R12 oldval, iRegI_R13 newval,
+instruct compareAndExchangeB(iRegINoSp res, indirect mem, iRegI_R12 oldval, iRegI_R13 newval, rFlagsReg cr)
```

Comment:
Similar here. No need to make `oldval` or `newval` fixed registers for `compareAndExchangeB`, `compareAndExchangeS`, `compareAndExchangeBAcq` or `compareAndExchangeSAcq`.
And we should also remove `cr` which is not used.

Reply:
Seems the Acq variants is missed? I see `compareAndSwapBAcq`, `compareAndSwapSAcq`, `compareAndExchangeBAcq` and `compareAndExchangeSAcq` still uses fixed registers for oldval and newval.